lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -1360,6 +1404,73 @@ instance (PassphraseMaxLength purpose, PassphraseMinLength purpose)
 instance ToJSON (ApiT (Passphrase purpose)) where
     toJSON = toJSON . toText . getApiT
 
+instance FromJSON ApiScript where
+    parseJSON =  withObject "ApiScript" $ \o ->
+        ApiScript . ApiT <$> o .: "script"
+instance ToJSON ApiScript where
+    toJSON (ApiScript (ApiT script')) =
+        object ["script" .= toJSON script']
+
+instance FromJSON ApiPubKey where
+    parseJSON = parseJSON >=> eitherToParser . first ShowFmt . fromText
+instance ToJSON ApiPubKey where
+    toJSON (ApiPubKey key') = do
+        let hrp = [Bech32.humanReadablePart|addr_vk|]
+        String $ T.decodeUtf8 $ encode (EBech32 hrp) key'
+
+instance FromJSON ApiCredential where
+    parseJSON v =
+        (CredentialScript <$> parseJSON v) <|>
+        (CredentialPubKey <$> parsePubKey v)
+      where
+        parsePubKey = parseFromText "ApiCredential" "pub_key"
+
+parseFromText :: FromText a => String -> Text -> Aeson.Value -> Aeson.Parser a
+parseFromText typeName k = withObject typeName $ \o ->
+    fromText <$> o .: k >>= \case
+        Right bytes -> pure bytes
+        Left (TextDecodingError err) -> fail err
+
+instance ToJSON ApiCredential where
+    toJSON (CredentialPubKey c) = object ["pub_key" .= c]
+    toJSON (CredentialScript c) = toJSON c
+
+instance FromJSON ApiAddressData where
+    parseJSON v =
+        parseBaseAddr v <|>
+        parseEnterprise v <|>
+        parseRewardAccount v <|>
+        fail "ApiAddressData must have at least one credential."
+      where
+         parseBaseAddr = withObject "AddrBase" $ \o ->
+             AddrBase <$> o .: "spending" <*> o .: "staking"
+         parseEnterprise = withObject "AddrEnterprise" $ \o ->
+             AddrEnterprise <$> o .: "spending"
+         parseRewardAccount = withObject "AddrRewardAccount" $ \o ->
+             AddrRewardAccount <$> o .: "staking"
```

Comment:
I don't think we should use the word "staking" here or anywhere.
"Stake address" is ok, but not "staking address".
The verb is to delegate.

Reply:
I somewhat followed the naming from https://hydra.iohk.io/build/4598683/download/1/delegation_design_spec.pdf - see Fig 2. There is Payment Credential and Stake Credential...so I used spending/staking... You think spending/delegating is better right?

Reply:
please notice that `ApiAddressData` has data containing credentials used either for spending or staking or both...

Reply:
I think use "payment" and "stake" here then. The word "staking" does not exist in the spec.

Reply:
ok, will change to spending/delegating

Reply:
done in https://github.com/input-output-hk/cardano-wallet/pull/2253/commits/3d003584ca7245da6c5f7012b712a4880d7955be